### PR TITLE
PWN-8637 - Striga. Feature toggle and design fixes

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-8637

## Description of Work
- banner fixes
- home action bottom sheet changed
- top up done with feature-toggle and presenter

- local testing tbd

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/jupiter/repository/routes/JupiterSwapRouteValidator.kt [hunk 1]
```diff
@@ -38,9 +38,9 @@ class JupiterSwapRouteValidator(
     suspend fun validateRoutes(
         routes: List<JupiterSwapRoute>,
     ): List<JupiterSwapRoute> = withContext(dispatchers.io) {
-        if (!swapRoutesValidationEnabled.isFeatureEnabled) {
-            return@withContext routes
-        }
+//        if (!swapRoutesValidationEnabled.isFeatureEnabled) {
+//            return@withContext routes
+//        }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/jupiter/repository/routes/JupiterSwapRouteValidator.kt:43]:
```suggestion
        if (!swapRoutesValidationEnabled.isFeatureEnabled) {
            return@withContext routes
        }
```

File: app/src/main/java/org/p2p/wallet/moonpay/ui/BuyFragmentFactory.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import androidx.fragment.app.Fragment
+import org.p2p.core.token.Token
+import org.p2p.wallet.common.feature_toggles.toggles.remote.NewBuyFeatureToggle
+import org.p2p.wallet.moonpay.ui.new.NewBuyFragment
+
+class BuyFragmentFactory(
+    private val buyFeatureToggle: NewBuyFeatureToggle
+) {
+
+    fun buyFragment(tokenForBuy: Token): Fragment {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/moonpay/ui/BuyFragmentFactory.kt:12]: Replace all checks with this method

File: app/src/main/java/org/p2p/wallet/striga/ui/TopUpWalletPresenter.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+class TopUpWalletPresenter(
+    private val userInteractor: UserInteractor,
+    private val strigaSignupFeatureToggle: StrigaSignupEnabledFeatureToggle,
+) : BasePresenter<TopUpWalletContract.View>(),
+    TopUpWalletContract.Presenter {
+
+    override fun attach(view: TopUpWalletContract.View) {
+        super.attach(view)
+
+        if (strigaSignupFeatureToggle.isFeatureEnabled) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/striga/ui/TopUpWalletPresenter.kt:17]: Add a check for web3auth too

File: app/src/main/java/org/p2p/wallet/striga/ui/TopUpWalletPresenter.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+) : BasePresenter<TopUpWalletContract.View>(),
+    TopUpWalletContract.Presenter {
+
+    private val isUserAuthByWeb3: Boolean
+        get() = seedPhraseProvider.getUserSeedPhrase().provider == SeedPhraseSource.WEB_AUTH
+
+    override fun attach(view: TopUpWalletContract.View) {
+        super.attach(view)
+
+        if (strigaSignupFeatureToggle.isFeatureEnabled && isUserAuthByWeb3) {
```
Comment (Davran331122) [inline app/src/main/java/org/p2p/wallet/striga/ui/TopUpWalletPresenter.kt:23]: `val isStrigaEnabled = strigaSignupFeatureToggle.isFeatureEnabled && isUserAuthByWeb3`
`view?.setStrigaBankTransferVisibility(isVisible = isStrigaEnabled)`

**Review by Davran331122 — APPROVED** [review]

**Review by kabramovp2p — APPROVED** [review]

**Pull request merged**